### ISSUE TITLE
Error when viewing entry revisions with nested Vizy fields

### ISSUE THREAD

**Issue by alexroper** [verbb/vizy#154]
**Description**

When an entry has a Vizy field/block nested inside another Vizy field, viewing previous entry revisions causes this error:

```
An Error occurred while handling another error:
TypeError: verbb\vizy\nodes\VizyBlock::renderStaticHtml(): Return value must be of type ?Twig\Markup, string returned in /app/vendor/verbb/vizy/src/nodes/VizyBlock.php:178
Stack trace:
#0 /app/vendor/verbb/vizy/src/models/NodeCollection.php(132): verbb\vizy\nodes\VizyBlock->renderStaticHtml()
#1 /app/vendor/verbb/vizy/src/fields/VizyField.php(299): verbb\vizy\models\NodeCollection->renderStaticHtml()
#2 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(241): verbb\vizy\fields\VizyField->getStaticHtml(Object(verbb\vizy\models\NodeCollection), Object(craft\elements\MatrixBlock))
#3 /app/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\MatrixBlock), true)
#4 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\MatrixBlock), true)
#5 /app/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#6 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#7 /app/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\MatrixBlock), true)
#8 /app/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#9 /app/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#10 /app/vendor/twig/twig/src/Extension/CoreExtension.php(1607): craft\models\FieldLayout->createForm(Object(craft\elements\MatrixBlock), true)
#11 /app/vendor/craftcms/cms/src/helpers/Template.php(101): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\models\FieldLayout), 'createForm', Array, 'method', false, false)
#12 /app/storage/runtime/compiled_templates/2d/2d38e39d4bc46ef982c09193ef3d3e9a.php(293): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\models\FieldLayout), 'createForm', Array, 'method')
#13 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_50a1e5b4999d8ac1ece7b7c3931abbf2->doDisplay(Array, Array)
#14 /app/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#15 /app/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#16 /app/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\Template->render(Array, Array)
#17 /app/vendor/twig/twig/src/Environment.php(277): Twig\TemplateWrapper->render(Array)
#18 /app/vendor/craftcms/cms/src/web/View.php(414): Twig\Environment->render('_components/fie...', Array)
#19 /app/vendor/craftcms/cms/src/fields/Matrix.php(858): craft\web\View->renderTemplate('_components/fie...', Array)
#20 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(241): craft\fields\Matrix->getStaticHtml(Array, Object(craft\elements\Entry))
#21 /app/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\Entry), true)
#22 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\Entry), true)
#23 /app/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#24 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#25 /app/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\Entry), true)
#26 /app/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#27 /app/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#28 /app/vendor/craftcms/cms/src/controllers/ElementsController.php(720): craft\models\FieldLayout->createForm(Object(craft\elements\Entry), true, Array)
#29 /app/vendor/craftcms/cms/src/controllers/ElementsController.php(426): craft\controllers\ElementsController->_prepareEditor(Object(craft\elements\Entry), false, Object(craft\web\Response), 'main-form', Object(Closure), Object(Closure), Object(Closure))
#30 [internal function]: craft\controllers\ElementsController->craft\controllers\{closure}(Object(craft\web\Response), 'main-form')
#31 /app/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(108): call_user_func(Object(Closure), Object(craft\web\Response), 'main-form')
#32 /app/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(48): craft\web\CpScreenResponseFormatter->_formatTemplate(Object(craft\web\Response), Object(craft\web\CpScreenResponseBehavior))
#33 /app/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\CpScreenResponseFormatter->format(Object(craft\web\Response))
#34 /app/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#35 /app/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#36 /app/vendor/yiisoft/yii2/web/ErrorHandler.php(136): yii\web\Response->send()
#37 /app/vendor/craftcms/cms/src/web/ErrorHandler.php(194): yii\web\ErrorHandler->renderException(Object(TypeError))
#38 /app/vendor/yiisoft/yii2/base/ErrorHandler.php(135): craft\web\ErrorHandler->renderException(Object(TypeError))
#39 /app/vendor/craftcms/cms/src/web/ErrorHandler.php(64): yii\base\ErrorHandler->handleException(Object(TypeError))
#40 [internal function]: craft\web\ErrorHandler->handleException(Object(TypeError))
#41 {main}
Previous exception:
TypeError: verbb\vizy\nodes\VizyBlock::renderStaticHtml(): Return value must be of type ?Twig\Markup, string returned in /app/vendor/verbb/vizy/src/nodes/VizyBlock.php:178
Stack trace:
#0 /app/vendor/verbb/vizy/src/models/NodeCollection.php(132): verbb\vizy\nodes\VizyBlock->renderStaticHtml()
#1 /app/vendor/verbb/vizy/src/fields/VizyField.php(299): verbb\vizy\models\NodeCollection->renderStaticHtml()
#2 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(241): verbb\vizy\fields\VizyField->getStaticHtml(Object(verbb\vizy\models\NodeCollection), Object(craft\elements\MatrixBlock))
#3 /app/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\MatrixBlock), true)
#4 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\MatrixBlock), true)
#5 /app/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#6 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#7 /app/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\MatrixBlock), true)
#8 /app/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#9 /app/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#10 /app/vendor/twig/twig/src/Extension/CoreExtension.php(1607): craft\models\FieldLayout->createForm(Object(craft\elements\MatrixBlock), true)
#11 /app/vendor/craftcms/cms/src/helpers/Template.php(101): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\models\FieldLayout), 'createForm', Array, 'method', false, false)
#12 /app/storage/runtime/compiled_templates/2d/2d38e39d4bc46ef982c09193ef3d3e9a.php(293): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\models\FieldLayout), 'createForm', Array, 'method')
#13 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_50a1e5b4999d8ac1ece7b7c3931abbf2->doDisplay(Array, Array)
#14 /app/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#15 /app/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#16 /app/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\Template->render(Array, Array)
#17 /app/vendor/twig/twig/src/Environment.php(277): Twig\TemplateWrapper->render(Array)
#18 /app/vendor/craftcms/cms/src/web/View.php(414): Twig\Environment->render('_components/fie...', Array)
#19 /app/vendor/craftcms/cms/src/fields/Matrix.php(858): craft\web\View->renderTemplate('_components/fie...', Array)
#20 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(241): craft\fields\Matrix->getStaticHtml(Array, Object(craft\elements\Entry))
#21 /app/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\Entry), true)
#22 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(210): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\Entry), true)
#23 /app/vendor/craftcms/cms/src/web/View.php(1544): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#24 /app/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(211): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#25 /app/vendor/craftcms/cms/src/models/FieldLayout.php(681): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\Entry), true)
#26 /app/vendor/craftcms/cms/src/web/View.php(1539): craft\models\FieldLayout->craft\models\{closure}()
#27 /app/vendor/craftcms/cms/src/models/FieldLayout.php(680): craft\web\View->namespaceInputs(Object(Closure), NULL)
#28 /app/vendor/craftcms/cms/src/controllers/ElementsController.php(720): craft\models\FieldLayout->createForm(Object(craft\elements\Entry), true, Array)
#29 /app/vendor/craftcms/cms/src/controllers/ElementsController.php(426): craft\controllers\ElementsController->_prepareEditor(Object(craft\elements\Entry), false, Object(craft\web\Response), 'main-form', Object(Closure), Object(Closure), Object(Closure))
#30 [internal function]: craft\controllers\ElementsController->craft\controllers\{closure}(Object(craft\web\Response), 'main-form')
#31 /app/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(108): call_user_func(Object(Closure), Object(craft\web\Response), 'main-form')
#32 /app/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(48): craft\web\CpScreenResponseFormatter->_formatTemplate(Object(craft\web\Response), Object(craft\web\CpScreenResponseBehavior))
#33 /app/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\CpScreenResponseFormatter->format(Object(craft\web\Response))
#34 /app/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#35 /app/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#36 /app/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#37 /app/web/index.php(12): yii\base\Application->run()
#38 {main}
```

**Steps to reproduce**
1. Create a Vizy field with a block type that includes another Vizy field.
2. Add this Vizy field to a section and create an entry in the section.
3. Add content to the Vizy field that includes content with the nested Vizy field/block type.
4. Save a couple different changes to the entry content.
5. View a previous revision of the content.

<img width="348" alt="Screen Shot 2022-06-10 at 10 12 53 AM" src="https://user-images.githubusercontent.com/377542/173117641-26982f20-eebd-4258-9c9d-472b95712c5e.png">

**Additional info**
- Plugin version: 2.0.2
- Craft version: Craft Pro 4.0.4
- Multi-site: No

**Additional context**

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/vizy` requirement in `composer.json` to:

```
"require": {
  "verbb/vizy": "dev-craft-4 as 2.0.2",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.0.3](https://github.com/verbb/vizy/releases/tag/2.0.3)
